core/src/snapshot_packager_service.rs

```diff
@@ -93,25 +93,48 @@ impl SnapshotPackagerService {
                         });
                     }
 
+                    let archive_time = Instant::now();
+                    // Serializing the snapshot package is not allowed to fail, as archiving is
+                    // not allowed to fail (see comment on archive_snapshot_package below
+                    let bank_snapshot_info = snapshot_utils::serialize_snapshot(
+                        &snapshot_config.bank_snapshots_dir,
+                        snapshot_config.snapshot_version,
+                        snapshot_package.bank_snapshot_package,
+                        snapshot_package.snapshot_storages.as_slice(),
+                        exit_backpressure.is_none(),
+                    );
+
+                    let Ok(bank_snapshot_info) = bank_snapshot_info else {
+                        error!(
+                            "Stopping {}! Fatal error while serializing snapshot for slot {}: \
+                             {bank_snapshot_info:?}",
```

Comment:
Can we get the inner error out from this `Result::Error`, please? Then log it with Display instead of Debug.

I also thought clippy would require inlining the `snapshot_slot` var into the format string here? I guess not?